common/cards/season-x/hermits/skizzleman-rare.ts

```diff
@@ -33,58 +34,54 @@ class SkizzlemanRare extends Card {
 		},
 	}
 
+	newGasLightAttack(component: CardComponent, target: RowEntity) {
+		return {
+			attacker: component.entity,
+			target: target,
+			type: 'secondary',
+			log: (values) => `${values.target} took ${values.damage} damage from $vGas Light$`,
+		} satisfies AttackDefs
+	}
+
 	override onAttach(game: GameModel, component: CardComponent, observer: ObserverComponent) {
-		const {player} = component
+		const {player, opponentPlayer} = component
 
 		let attackedEntities = new Set<RowEntity>()
 		let usedSecondary = false
 
+		observer.subscribe(player.hooks.onTurnStart, () => {
+			attackedEntities = new Set()
+			usedSecondary = false
+		})
+
 		observer.subscribe(player.hooks.onAttack, (attack) => {
 			if (attack.isAttacker(component.entity) && attack.type === 'secondary') {
 				usedSecondary = true
 			}
+		})
 
+		observer.subscribe(opponentPlayer.hooks.onDefence, (attack) => {
 			// Status effect attacks have a special case because they happen at the end of the turn
-			if (attack.type === 'status-effect') {
-				if (attack.targetEntity && attackedEntities.has(attack.targetEntity)) return
-				attack.addDamage(component.entity, 20)
+			if (attack.type === 'status-effect' && attack.targetEntity) {
+				if (attackedEntities.has(attack.targetEntity)) return
+				attack.addNewAttack(game.newAttack(this.newGasLightAttack(component, attack.targetEntity)))
 				return
 			}
 
-			if (!attack.targetEntity) return
-			if (
-				!game.components.exists(
-					RowComponent,
-					query.row.entity(attack.targetEntity),
-					query.row.opponentPlayer,
-					query.not(query.row.active)
-				)
-			) {
-				return
-			}
-
-			attackedEntities.add(attack.targetEntity)
+			if (attack.targetEntity) attackedEntities.add(attack.targetEntity)
 		})
 
 		observer.subscribe(player.hooks.onTurnEnd, () => {
```

Comment:
Does not filter out the active Hermit and does not work when "Gas Light" is mocked

Reply:
I fixed this by changing mocked cards to delete themselves at the end of the turn instead of after attack.

Reply:
This logic should probably be moved to a status effect since `onTurnEnd` will be removed if Skizz is knocked out by Thorns/Wolf

Reply:
I agree with that. I will convert this over to a status effect.